var/spack/repos/builtin/packages/netpbm/package.py

```diff
@@ -0,0 +1,207 @@
+            cflags = ['-O3', '-ffast-math', '-pedantic', '-Wall', '-Wimplicit']
+            cflags.extend(['-Wno-uninitialized', '-Wmissing-declarations'])
+            cflags.extend(['-Wwrite-strings', '-Wmissing-prototypes'])
+            cflags.extend(['-Wundef', '-Wno-unknown-pragmas'])
+            if 'platform=darwin' in spec:
+                # https://github.com/macports/macports-ports/blob/master/graphics/netpbm/Portfile
+                cflags.append('-D_DARWIN_C_SOURCE')
+                # http://www.linuxquestions.org/questions/linux-from-scratch-13/can't-compile-luit-xorg-applications-4175476308/
+                # cflags.append('-U_XOPEN_SOURCE')
+                # http://www.mistys-internet.website/blog/blog/2013/10/19/no-cpp-precomp-the-compiler-flag-that-time-forgot/
+                # cflags.append('-no-cpp-precomp')
+            config.append('CFLAGS = {0}'.format(' '.join(cflags)))
+            config.append('CFLAGS_SHLIB += -fno-common')
+
+        if '+all' in spec:
+            flex = join_path(spec['flex'].prefix.bin, 'flex')
+            if os.path.exists(flex):
+                config.append('LEX = {0}'.format(flex))
+
+        config.append('TIFFLIB={0}'.format(
+            spec['libtiff'].libs.ld_flags))
+        config.append('TIFFHDR_DIR={0}'.format(
+            spec['libtiff'].headers.directories[0]))
+        config.append('PNGLIB={0}'.format(
+            spec['libpng'].libs.ld_flags))
+        config.append('PNGHDR_DIR={0}'.format(
+            spec['libpng'].headers.directories[0]))
+        config.append('JPEGLIB={0}'.format(
+            spec['jpeg'].libs.ld_flags))
+        config.append('JPEGHDR_DIR={0}'.format(
+            spec['jpeg'].headers.directories[0]))
+        if '+all' in spec and '+builtin' not in spec:
+            config.append('JASPERLIB={0}'.format(
+                spec['jasper'].libs.ld_flags))
+            config.append('JASPERHDR_DIR={0}'.format(
+                spec['jasper'].headers.directories[0]))
+            config.append('JBIGLIB={0}'.format(
+                spec['jbigkit'].libs.ld_flags))
+            config.append('JBIGHDR_DIR={0}'.format(
+                spec['jbigkit'].headers.directories[0]))
+        if '+X' in spec:
+            pkg_config = which('pkg-config')
+            if not pkg_config('x11', '--exists'):
+                config.append('X11LIB={0}'.format(
+                    spec['libx11'].libs.ld_flags))
+                config.append('X11HDR_DIR={0}'.format(
+                    spec['libx11'].headers.directories[0]))
+                config.append('ZLIB={0}'.format(spec['zlib'].libs.ld_flags))
+        config.append('NETPBM_DOCURL = http://netpbm.sourceforge.net/doc/')
+        if spec.target.family == 'x86_64':
+            config.append('WANT_SSE = Y')
+
+        with open('config.mk', 'a') as mk:
+            mk.write('\n'.join(config))
+
+    def build(self, spec, prefix):
+        make()
+
+    def install(self, spec, prefix):
```

Comment:
There is no make install available?

Reply:
No, there is a perl script buildtools/installnetpbm.pl that is called from ./installnetpbm
alike for configure.
Another option would be to make these perl scripts batch capable (as opposed to purely interactive and talkative) but I found easier to reimplement the stuff in the spack package.
Is it inconvenient for spack to have so long a package ?

Reply:
The only issue could be that the file structure changes between versions, so you'd have to change the package.py to support multiple versions at the same time, which isn't great. But we can decide on that when they actually change what they install.

Reply:
Absolutely right, I know it is rather fragile. The good point is that Bryan Henderson the maintainer of netpbm is very reactive.